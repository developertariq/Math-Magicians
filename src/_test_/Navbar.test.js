import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct home page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const homeLink = getByText(/Home/i);
    fireEvent.click(homeLink);
    expect(window.location.pathname).toBe('/');
  });

  test('renders the home page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const homeLink = getByText(/Home/i);
    fireEvent.click(homeLink);
    expect(homeLink.classList.contains('active')).toBe(true);
  });

  test('renders the correct calculator page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const calcLink = getByText(/Calculator/i);
    fireEvent.click(calcLink);
    expect(window.location.pathname).toBe('/Calculator');
  });

  test('renders the calculator page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const calcLink = getByText(/Calculator/i);
    fireEvent.click(calcLink);
    expect(calcLink.classList.contains('active')).toBe(true);
  });

  test('renders the correct quote page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const quoteLink = getByText(/Quote/i);
    fireEvent.click(quoteLink);
    expect(window.location.pathname).toBe('/Quote');
  });

  test('renders the quote page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const quoteLink = getByText(/Quote/i);
    fireEvent.click(quoteLink);
    expect(quoteLink.classList.contains('active')).toBe(true);
  });
});
