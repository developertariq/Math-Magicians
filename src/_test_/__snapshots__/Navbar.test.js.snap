// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navBar"
  data-testid="navbar"
>
  <ul
    className="menuNav"
  >
    <li>
      <a
        activeClassName="active-link"
        aria-current="page"
        className="active"
        exact={true}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        className=""
        exact={true}
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        className=""
        exact={true}
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
